.github/workflows/example_android_integration_test.yml

```diff
@@ -0,0 +1,40 @@
+name: Example Android integration test
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+env:
+  FLUTTER_VERSION: "3.3.10"
+  EXAMPLE_DIRECTORY: example
+
+jobs:
+  integration_test_android:
+    runs-on: macos-12
+    steps:
+      - uses: actions/checkout@v3
+      - uses: subosito/flutter-action@v2
+        with:
+          flutter-version: ${{ env.FLUTTER_VERSION }}
+          channel: stable
+      - uses: actions/setup-java@v3
+        with:
+          java-version: "11"
+          distribution: "temurin"
+          cache: gradle
+      - run: flutter doctor -v
+      - run: flutter pub get
+
+      - uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: 29
+          arch: x86_64
+          script: cd example && flutter drive --target=test_driver/app.dart
+
+      - uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: my-artifact
+          path: ./example/screenshots/*
```

Comment:
If you already have an environment variable, you might want to replace this with ` ${{ env.EXAMPLE_DIRECTORY }}`

Reply:
Ah I see that you do this in the other files already. :)

Reply:
Ah ok I read about it 😅
For environment variables created in other steps, make sure to use the `env` expression syntax
```yaml
    steps:
    - run: | 
        mkdir testing
        echo "This is a file to upload" > testing/file.txt
        echo "artifactPath=testing/file.txt" >> $GITHUB_ENV
    - uses: actions/upload-artifact@v3
      with:
        name: artifact
        path: ${{ env.artifactPath }} # this will resolve to testing/file.txt at runtime
```

Reply:
Yes, exactly! :D